src/model/services/mapi/capiv1alpha3/key.ts

```diff
@@ -41,26 +42,32 @@ export function getKubernetesAPIEndpointURL(
   return `https://${hostname}`;
 }
 
+interface IConditionGetter {
+  status?: {
+    conditions?: ICondition[];
+  };
+}
+
 export function getCondition(
```

Comment:
Thanks to duck typing we can make these condition helpers more generic, and use them for multiple resources